src/components/schedule-visualizer/tests/colors.test.tsx

```diff
@@ -0,0 +1,33 @@
+import ColorManager from '../colors';
+
+describe('Color manager', () => {
+  let colorManager: ColorManager;
+  const mockRobot = {
+    name: 'robot',
+    model: 'A',
+  };
+
+  beforeEach(() => {
+    colorManager = new ColorManager();
+  });
+
+  it('should store and return a color when robotColor is called', async () => {
+    const colorPromise = await colorManager.robotColor(mockRobot.name, mockRobot.model);
+    expect(colorPromise).toEqual(colorManager.robotColorFromCache(mockRobot.name));
+  });
+
+  it('should store and return a color when robotTrajectoryColor is called', async () => {
+    const colorPromise = await colorManager.robotTrajectoryColor(mockRobot.name, mockRobot.model);
+    expect(colorPromise).toEqual(colorManager.pathColorFromCache(mockRobot.name));
+  });
+
+  it('should return a promise when robotImageColor is called', done => {
```

Comment:
Did you test this? https://create-react-app.dev/docs/adding-images-fonts-and-files/
Not sure if it will work for unit tests.

Reply:
Tried importing the icons in the public assets but seems like it is out of scope